### PR TITLE
Test: Harden lint-filenames.sh

### PR DESCRIPTION
**Pull request by ghost**
- This fixes that only files with lower case file name suffix where found before, which contradicted the Regex to find uppercase characters in file names (including suffixes I guess).
- `--full-name` switch was added to git ls-files, to define that the found file always includes it's full path in the git project.
- since we know now that the file name includes the full path, we can harden the Regex to exclude the  secp256k1 and univalue sub folders.
- use backslash line break to make code easier to read and avoid too long line.

### CODE REVIEW

File: test/lint/lint-filenames.sh [hunk 1]
```diff
@@ -10,7 +10,11 @@
 export LC_ALL=C
 
 EXIT_CODE=0
-OUTPUT=$(git ls-files -- "*.cpp" "*.h" "*.py" "*.sh" | grep -vE '^[a-z0-9_./-]+$' | grep -vE 'src/(secp256k1|univalue)/')
+OUTPUT=$(git ls-files --full-name -- "*.[cC][pP][pP]" "*.[hH]" "*.[pP][yY]" "*.[sS][hH]" \
+| grep -vE '^[a-z0-9_./-]+$' \
+| grep -vE '^src/(secp256k1|univalue)/' \
+)
```
Comment (practicalswift) [inline test/lint/lint-filenames.sh]: Nit: Please format it like this to make it consistent with the rest of the lint shell scripts:

```
OUTPUT=$(git ls-files --full-name -- "*.[cC][pP][pP]" "*.[hH]" "*.[pP][yY]" "*.[sS][hH]" | \
    grep -vE '^[a-z0-9_./-]+$' | \
    grep -vE '^src/(secp256k1|univalue)/')
```

**Comment by practicalswift** [comment]
utACK 8fd05107ee5c188c9ab59b0c1b3f5698f8075c8d modulo nits

Luckily:

```
$ git log -u | grep -E '^\+\+\+ b/' | grep -E '\.(CPP|H|PY|SH)$'
$
```

**Comment by wodry** [comment]
Changed coding style, allthough I think to have the closing bracket on it's own line would be clearer to read and less error prone if a line should be attached, also better diff reading. But that's no discussion for a single PR, so changed according to the existing scripting stle as requested. Rebased and squashed.

**Comment by practicalswift** [comment]
ACK 927e1150bc207181f7d5b0948e87d04b737a1c27

**Comment by wodry** [comment]
Yes, only lowercase suffixes, here the Regex that also would find mixed case suffixes:
```
git log -u | grep -E '^\+\+\+ b/' | grep -iE '\.(cpp|h|py|sh)$' | grep -cvE '\.(cpp|h|py|sh)$'
0
```

**Review by ken2812221 — APPROVED** [review]
utACK 927e115

**Pull request merged**